weather_mv/loader_pipeline/bq.py

```diff
@@ -253,57 +329,69 @@ def extract_rows(self, uri: str, coordinates: t.List[t.Dict]) -> t.Iterator[t.Di
         with open_dataset(uri, self.xarray_open_dataset_kwargs, self.disable_grib_schema_normalization,
                           self.tif_metadata_for_start_time, self.tif_metadata_for_end_time, is_zarr=self.zarr) as ds:
             data_ds: xr.Dataset = _only_target_vars(ds, self.variables)
-            yield from self.to_rows(coordinates, data_ds, uri)
+            if self.area:
+                n, w, s, e = self.area
+                data_ds = data_ds.sel(latitude=slice(n, s), longitude=slice(w, e))
+                logger.info(f'Data filtered by area, size: {data_ds.nbytes}')
+            yield from self.to_rows(coordinate, data_ds, uri)
 
-    def to_rows(self, coordinates: t.Iterable[t.Dict], ds: xr.Dataset, uri: str) -> t.Iterator[t.Dict]:
+    def to_rows(self, coordinate: t.Dict, ds: xr.Dataset, uri: str) -> t.Iterator[t.Dict]:
         first_ts_raw = (
             ds.time[0].values if isinstance(ds.time.values, np.ndarray)
             else ds.time.values
         )
         first_time_step = to_json_serializable_type(first_ts_raw)
-        for it in coordinates:
-            # Use those index values to select a Dataset containing one row of data.
-            row_ds = ds.loc[it]
+        with open_local(self.geo_data_csv_path) as master_lat_lon:
+            selected_ds = ds.loc[coordinate]
 
-            # Create a Name-Value map for data columns. Result looks like:
-            # {'d': -2.0187, 'cc': 0.007812, 'z': 50049.8, 'rr': None}
-            row = {n: to_json_serializable_type(ensure_us_time_resolution(v.values))
-                   for n, v in row_ds.data_vars.items()}
+            # Ensure that the latitude and longitude dimensions are in sync with the geo data CSV.
+            if not BQ_EXCLUDE_COORDS - set(selected_ds.dims.keys()):
+                selected_ds = selected_ds.transpose('latitude', 'longitude')
 
-            # Serialize coordinates.
-            it = {k: to_json_serializable_type(v) for k, v in it.items()}
+            master_df = pd.read_csv(master_lat_lon)
```

Comment:
Let’s use a different term from “master”.

Reply:
How about `vector_df`, `vector_data` or `rows_data` ? any other suggestion ?

Reply:
I like the first two. How about `vector_df`?

Reply:
Yes, this sounds good.  Necessary code changes done.